### PR TITLE
small fixes for new test harness using server engine

### PR DESCRIPTION
**Pull request by jennifersp**
This PR adds small fixes to server_engine that is used to test existing engine tests over running server.

### CODE REVIEW

File: enginetest/evaluation.go
Comment (fulghum) [inline enginetest/evaluation.go]: It would be good to document a little bit more about why we're doing this and how we could remove this in the future. You mention that it's for the server tests, but someone reading this in a year will probably need more context. 

Ideally, we wouldn't need to massage the results coming back like this, so I think if we're going to add in code like this, we should explain why and ideally add a TODO that explains what we need to do to remove this special case. For this one in particular, it seems like we either have serialization code in the `SQL` functions that translates booleans to a 1 or 0 when we send them over the wire, or... maybe it's the go-mysql-driver that's showing us different results? Either way, would be good to explain where it's coming from and what we'd have to do to clean it up in the future.

File: enginetest/queries/variable_queries.go
Comment (fulghum) [inline enginetest/queries/variable_queries.go:337]: Does MySQL treat this as a string, too? I'm surprised that having a decimal in the value would cause the user var to be treated as a string.

File: enginetest/server_engine.go
Comment (fulghum) [inline enginetest/server_engine.go:231]: Looks like this is just leftover debugging code?

File: enginetest/queries/variable_queries.go
Comment (fulghum) [reply enginetest/queries/variable_queries.go:337]: ooooh... actually... I bet this is because it's a decimal and our tests turn decimal values into strings for easier comparison, right?

**Review by fulghum — APPROVED** [review]
Looks good. Just a few minor comments/questions. 

Probably smart to do a Dolt CI test run before merging this and kicking off the auto-bump, just so you can have any test updates identified and ready to go before the GMS -> Dolt queue gets blocked by a failing bump.

File: enginetest/queries/variable_queries.go
Comment (jennifersp) [reply enginetest/queries/variable_queries.go:337]: yep! 🙂

**Pull request merged**
